dev-python/pytest-describe/pytest-describe-0.11.1.ebuild

```diff
@@ -0,0 +1,28 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="7"
+
+PYTHON_COMPAT=( python2_7 python3_{4,5,6,7} )
+
+inherit distutils-r1
+
+DESCRIPTION="Describe-style plugin for py.test"
+HOMEPAGE="https://github.com/ropez/${PN} https://pypi.org/project/${PN}"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~ia64 ~ppc ~ppc64 ~sparc ~x86"
+IUSE="test"
+RESTRICT="!test? ( test )"
+
+RDEPEND=">=dev-python/pytest-2.6.0[${PYTHON_USEDEP}]"
+
+DEPEND="
+	dev-python/setuptools[${PYTHON_USEDEP}]
+	test? ( >=dev-python/pytest-2.6.0[${PYTHON_USEDEP}] )"
+
+python_test() {
+	pytest -vv || die "Tests failed under ${EPYTHON}"
```

Comment:
Tests fail for me:

```
=============================================================== test session starts ===============================================================
platform linux2 -- Python 2.7.15, pytest-3.6.3, py-1.5.3, pluggy-0.6.0 -- /usr/bin/python2.7
cachedir: .pytest_cache
rootdir: /tmp/portage/dev-python/pytest-describe-0.11.1/work/pytest-describe-0.11.1, inifile:
plugins: describe-0.11.1, xdist-1.22.2, virtualenv-1.2.11, shutil-1.2.11, mock-1.10.0, forked-0.2, hypothesis-3.59.1, flaky-3.4.0, backports.unittest-mock-1.3
collecting ... collected 21 items

test/test_collect.py::test_collect PASSED                                                                                                   [  4%]
test/test_collect.py::test_describe_evaluated_once PASSED                                                                                   [  9%]
test/test_custom_prefix.py::test_collect_custom_prefix FAILED                                                                               [ 14%]
test/test_fixtures.py::test_can_access_local_fixture PASSED                                                                                 [ 19%]
test/test_fixtures.py::test_can_access_fixture_from_nested_scope PASSED                                                                     [ 23%]
test/test_fixtures.py::test_local_fixture_overrides PASSED                                                                                  [ 28%]
test/test_marks.py::test_special_marks PASSED                                                                                               [ 33%]
test/test_marks.py::test_keywords PASSED                                                                                                    [ 38%]
test/test_marks.py::test_module_marks PASSED                                                                                                [ 42%]
test/test_marks.py::test_mark_at_describe_function PASSED                                                                                   [ 47%]
test/test_output.py::test_verbose_output PASSED                                                                                             [ 52%]
test/test_shared.py::test_shared_behavior PASSED                                                                                            [ 57%]
test/test_shared.py::test_multiple_shared_behaviors PASSED                                                                                  [ 61%]
test/test_shared.py::test_fixture PASSED                                                                                                    [ 66%]
test/test_shared.py::test_override_fixture PASSED                                                                                           [ 71%]
test/test_shared.py::test_name_mangling PASSED                                                                                              [ 76%]
test/test_shared.py::test_nested_name_mangling PASSED                                                                                       [ 80%]
test/test_shared.py::test_evaluated_once PASSED                                                                                             [ 85%]
test/test_simple_execution.py::test_can_pass PASSED                                                                                         [ 90%]
test/test_simple_execution.py::test_can_fail PASSED                                                                                         [ 95%]
test/test_simple_execution.py::test_can_fail_and_pass PASSED                                                                                [100%]

==================================================================== FAILURES =====================================================================
___________________________________________________________ test_collect_custom_prefix ____________________________________________________________

testdir = <Testdir local('/tmp/portage/dev-python/pytest-describe-0.11.1/temp/pytest-of-portage/pytest-0/test_collect_custom_prefix0')>

    def test_collect_custom_prefix(testdir):
        testdir.makeini(ini)
    
        a_dir = testdir.mkpydir('a_dir')
        a_dir.join('test_a.py').write(py.code.Source("""
            def foo_scope():
                def bar_context():
                    def passes():
                        pass
        """))
    
        result = testdir.runpytest('--collectonly')
        print(result.outlines)
>       assert _collect_result(result) == [
            "collected 1 item",
            "<Module 'a_dir/test_a.py'>",
            "  <DescribeBlock 'foo_scope'>",
            "    <DescribeBlock 'bar_context'>",
            "      <Function 'passes'>",
        ]
E       assert ["      <Func...t===', '', ''] == ['collected 1 ...on 'passes'>"]
E         At index 0 diff: u"      <Function 'passes'>" != 'collected 1 item'
E         Full diff:
E         - [u"      <Function 'passes'>", u'', u'===Flaky Test Report===', u'', u'']
E         + ['collected 1 item',
E         +  "<Module 'a_dir/test_a.py'>",
E         +  "  <DescribeBlock 'foo_scope'>",
E         +  "    <DescribeBlock 'bar_context'>",
E         +  "      <Function 'passes'>"]

../../../../work/pytest-describe-0.11.1/test/test_custom_prefix.py:37: AssertionError
-------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------
=============================================================== test session starts ===============================================================
platform linux2 -- Python 2.7.15, pytest-3.6.3, py-1.5.3, pluggy-0.6.0
rootdir: /tmp/portage/dev-python/pytest-describe-0.11.1/temp/pytest-of-portage/pytest-0/test_collect_custom_prefix0, inifile: tox.ini
plugins: describe-0.11.1, xdist-1.22.2, virtualenv-1.2.11, shutil-1.2.11, mock-1.10.0, forked-0.2, hypothesis-3.59.1, flaky-3.4.0, backports.unittest-mock-1.3
collected 1 item
<Module 'a_dir/test_a.py'>
  <DescribeBlock 'foo_scope'>
    <DescribeBlock 'bar_context'>
      <Function 'passes'>

===Flaky Test Report===


===End Flaky Test Report===
========================================================== no tests ran in 0.02 seconds ===========================================================
[u'=============================================================== test session starts ===============================================================', u'platform linux2 -- Python 2.7.15, pytest-3.6.3, py-1.5.3, pluggy-0.6.0', u'rootdir: /tmp/portage/dev-python/pytest-describe-0.11.1/temp/pytest-of-portage/pytest-0/test_collect_custom_prefix0, inifile: tox.ini', u'plugins: describe-0.11.1, xdist-1.22.2, virtualenv-1.2.11, shutil-1.2.11, mock-1.10.0, forked-0.2, hypothesis-3.59.1, flaky-3.4.0, backports.unittest-mock-1.3', u'collected 1 item', u"<Module 'a_dir/test_a.py'>", u"  <DescribeBlock 'foo_scope'>", u"    <DescribeBlock 'bar_context'>", u"      <Function 'passes'>", u'', u'===Flaky Test Report===', u'', u'', u'===End Flaky Test Report===', u'========================================================== no tests ran in 0.02 seconds ===========================================================', u'']
===Flaky Test Report===


===End Flaky Test Report===
======================================================= 1 failed, 20 passed in 4.10 seconds =======================================================
```

Looks like you need a strong blocker on `dev-python/flaky`.

Reply:
To avoid a blocker on dev-python/flaky I did
```Bash
# We need to disable flaky plugin because test_custom_prefix test validates
# pytest output and flaky plugin would appear unexpectedly
PYTEST_ADDOPTS="-p no:flaky" pytest -vv || die "Tests failed under ${EPYTHON}"
```